react/features/base/settings/reducer.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+
+import { APP_WILL_MOUNT } from '../../app';
+import JitsiMeetJS, { browser } from '../lib-jitsi-meet';
+import { ReducerRegistry } from '../redux';
+import { PersistenceRegistry } from '../storage';
+import { randomHexString } from '../util';
+
+import { SETTINGS_UPDATED } from './actionTypes';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * The default/initial redux state of the feature {@code base/settings}.
+ *
+ * @type Object
+ */
+const DEFAULT_STATE = {};
+
+const STORE_NAME = 'features/base/settings';
+
+/**
+ * Sets up the persistence of the feature {@code base/settings}.
+ *
+ * Note: There are some values inherited from the old Settings.js that needs
+ * session-based storage but not persisted storage, such as cameraDeviceId
+ * and micDeviceId so we utilize the selective storage functionality of the
+ * persistency engine.
+ */
+PersistenceRegistry.register(STORE_NAME, {
+    audioOutputDeviceId: true,
+    avatarID: true,
+    avatarURL: true,
+    cameraDeviceId: false,
```

Comment:
why are the device ids not persisted? they should!

Reply:
This was implemented this way (see store=false boolean in the original Settings.js code) and I didn't want to change it. Also please see previous discussion with Lenny above.

I'm happy to change this in a follow-up PR, but I don't want to alter behaviour in this one.

Reply:
Wait, so those were initially stored in session storage, then when this got moved to persistency that was "lost", right?

If this is the case, the previous behavior was broken by persistency.

Since this is audio devices, it means that reloading the page would mean your settings are gone, which is kinda bad. I'd rather have them remembered forever than forgotten all the time.

So, let's fix it since we are here. Please persist them.

Reply:
No, they were implemented the same way. There was a single variable that stored the value, and it got lost when the page was reloaded. See: https://github.com/jitsi/jitsi-meet/blob/4e36127dc7259e5bc591a948bef0d36daa00cd6c/modules/settings/Settings.js#L142

Reply:
I was thinking about it, and maybe it wasn't persisted to avoid cases when a device gets disconnected from the computer? Not sure, just thinking out loud.

Reply:
Let's ask the hive mind :-)